### PR TITLE
Poclr messages

### PR DESCRIPTION
**Pull request by RABijl**
some message related additions as part of the upstreaming process

### CODE REVIEW

File: lib/CL/devices/remote/communication.c [hunk 1]
```diff
@@ -1321,6 +1321,29 @@ traffic_monitor_pthread (void *arg)
   free (a);
   return NULL;
 }
+
+__attribute__ ((visibility ("default"))) void
```
Comment (pjaaskel) [inline lib/CL/devices/remote/communication.c:1325]: What's the attribute for?

**Review by pjaaskel — CHANGES_REQUESTED** [review]

File: lib/CL/devices/remote/communication.c (hunk 1 above)
Comment (RABijl) [reply lib/CL/devices/remote/communication.c:1325]: the rest of the file has the visibility hidden and this function is used by clients to get info about the amount of data being sent.

Comment (jansol) [reply lib/CL/devices/remote/communication.c:1325]: The functions that are actually meant to be part of the ABI are defined in the header with POCL_EXPORT but I didn't want to advertise this function like that originally when it was just for some internal testing. If we want to integrate this more "officially" the function should be declared in the corresponding header with the appropriate macros.

Comment (pjaaskel) [reply lib/CL/devices/remote/communication.c:1325]: I already discussed the PoCL-specific (clGetDeviceInfo device query) extension with Yash. I think that's the clean way to go. Pls discuss with @Yashvardhan-A

Comment (RABijl) [reply lib/CL/devices/remote/communication.c]: I've made the change by using clGetDeviceInfo

File: CMakeLists.txt [hunk 2]
```diff
@@ -304,7 +304,7 @@ option(ENABLE_EXAMPLES "Build examples. Defaults to ON" ON)
 
 option(ENABLE_RDMA "Enable usage of RDMA libraries for memory allocations. Requires libRDMAcm and libverbs" OFF)
 
-option(ENABLE_TRAFFIC_MONITOR "Enable network traffic monitoring & logging in remote device" OFF)
+option(ENABLE_TRAFFIC_MONITOR "Enable network traffic monitoring & logging in remote device" ${ENABLE_REMOTE_CLIENT})
```
Comment (pjaaskel) [inline CMakeLists.txt:307]: What's the reason making this optional? The data is accumulated unconditionally so there should not be additional runtime overheads here.

**Review by pjaaskel — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (RABijl) [reply CMakeLists.txt:307]: The code was originally written as an optional thing and spawns an extra thread for these stats. I've kept the option to turn it off if so desired.

Comment (pjaaskel) [reply CMakeLists.txt:307]: But does it impact the overhead. It's not disabling the collection of the status from what I see? I prefer not to have options that are not clearly useful as they clutter the code and also make it more difficult test the different (build) paths.

**Pull request merged**